Marlin/src/module/ft_motion/stepping.h

```diff
@@ -32,7 +32,11 @@ FORCE_INLINE constexpr uint32_t a_times_b_shift_16(const uint32_t a, const uint3
 }
 #define FTM_NEVER uint32_t(UINT16_MAX)                               // Reserved number to indicate "no ticks in this frame" (FRAME_TICKS_FP+1 would work too)
 constexpr uint32_t FRAME_TICKS = STEPPER_TIMER_RATE / FTM_FS;        // Timer ticks per frame (by default, 1kHz)
-constexpr uint32_t TICKS_BITS = __builtin_clzl(FRAME_TICKS + 1UL);   // Bits to represent the max value (duration of a frame, +1 one for FTM_NEVER).
+#if (MOTHERBOARD == BOARD_SIMULATED)
+  constexpr uint32_t TICKS_BITS = __builtin_clz(FRAME_TICKS + 1U);   // Bits to represent the max value (duration of a frame, +1 one for FTM_NEVER).
+#else
+  constexpr uint32_t TICKS_BITS = __builtin_clzl(FRAME_TICKS + 1UL); // Bits to represent the max value (duration of a frame, +1 one for FTM_NEVER).
+#endif
```

Comment:
FRAME_TICKS is already a uint32, the compiler should do the right thing
```suggestion
constexpr uint32_t TICKS_BITS = __builtin_clz(FRAME_TICKS + 1);   // Bits to represent the max value (duration of a frame, +1 one for FTM_NEVER).
```